release_testing/functional_tests/F04.jenkinsfile

```
@@ -1,135 +1,133 @@
-        string(defaultValue: "/mnt/tmp-disk/register-run-db", description: 'Path to register run db', name:'RegisterRunPath')
-        string(defaultValue: "", description: 'Overrides name of the register run db', name: 'RunId')
     }
 
     stages {
-        stage('Build') {
-            steps {
-                script {
-                    currentBuild.description = "Building on ${env.NODE_NAME}"
+        stage('Start parallel stage execution') {
+            parallel {
+                stage('F02 block processing (mainnet) 0-15m') {
+                    steps {
+                        build job: '/ReleaseTesting/FunctionalTests/F02/F02', parameters: [
+                            string(name: 'BlockFrom', value: "${BlockFrom}"),
+                            string(name: 'BlockTo', value: "15000000"),
+                            string(name: 'GenesisBlock', value: ""),
+                            string(name: 'CustomFlags', value: "--archive-query-rate ${ArchiveQueryRate}"),
+                            string(name: 'AidaVersion', value: "${AidaVersion}"),
+                            string(name: 'CarmenVersion', value: "${CarmenVersion}"),
+                            string(name: 'ToscaVersion', value: "${ToscaVersion}"),
+                            string(name: 'SonicVersion', value: "${SonicVersion}")
+                        ]
+                    }
                 }
 
-                checkout scmGit(
-                    branches: [[name: "${AidaVersion}"]],
-                    userRemoteConfigs: [[
-                        url: 'https://github.com/Fantom-foundation/Aida.git'
-                    ]]
-                )
-                sh "git submodule update --init --recursive"
-
-                dir('carmen') {
-                	checkout scmGit(
-                		branches: [[name: "${CarmenVersion}"]],
-                		userRemoteConfigs: [[url: 'https://github.com/Fantom-foundation/Carmen.git']]
-                	)
-
-                    sh "git submodule update --recursive"
+                stage('F02 block processing (mainnet) 15m-30m') {
+                    steps {
+                        build job: '/ReleaseTesting/FunctionalTests/F02/F02', parameters: [
+                            string(name: 'BlockFrom', value: "15000001"),
+                            string(name: 'BlockTo', value: "30000000"),
+                            string(name: 'GenesisBlock', value: "15000000"),
+                            string(name: 'CustomFlags', value: "--archive-query-rate ${ArchiveQueryRate}"),
+                            string(name: 'AidaVersion', value: "${AidaVersion}"),
+                            string(name: 'CarmenVersion', value: "${CarmenVersion}"),
+                            string(name: 'ToscaVersion', value: "${ToscaVersion}"),
+                            string(name: 'SonicVersion', value: "${SonicVersion}")
+                        ]
+                    }
                 }
```

```suggestion
                stage('F02 block processing (mainnet) 15m-30m') {
                    steps {
                        build job: '/ReleaseTesting/FunctionalTests/F02/F02', parameters: [
                            string(name: 'BlockFrom', value: "15000001"),
                            string(name: 'BlockTo', value: "30000000"),
                            string(name: 'GenesisBlock', value: "15000000"),
                            string(name: 'CustomFlags', value: "--archive-query-rate ${ArchiveQueryRate}"),
                            string(name: 'BucketPath', value: "carmen-genesis-generator/74"),
                            string(name: 'AidaVersion', value: "${AidaVersion}"),
                            string(name: 'CarmenVersion', value: "${CarmenVersion}"),
                            string(name: 'ToscaVersion', value: "${ToscaVersion}"),
                            string(name: 'SonicVersion', value: "${SonicVersion}")
                        ]
                    }
                }
```
I'm fully aware that you copied this from my code and it's my mistake.